packages/react/src/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,394 @@
+        .getPropertyValue('--brand-Tabs-container-padding')
+      const containerPadding = parseFloat(containerPaddingRem) * 16
+      const tabLeft = activeTabButton.offsetLeft - containerPadding
+      const tabWidth = activeTabButton.offsetWidth
+      const tabCenter = tabLeft + tabWidth / 2
+
+      const targetScrollLeft = tabCenter - containerWidth / 2
+
+      const maxScrollLeft = tabsContainer.scrollWidth - containerWidth // defines boudnary
+      const finalScrollLeft = Math.max(0, Math.min(targetScrollLeft, maxScrollLeft))
+
+      tabsContainer.scrollTo({
+        left: finalScrollLeft,
+        behavior: 'smooth',
+      })
+
+      return () => {
+        resizeObserver.disconnect()
+      }
+    }, [activeTab])
+
+    // Tracks the overflow state for mobile gradient visibility
+    // Should only appear when enough items trigger an overflow
+    // Related to gradient functionality
+    useEffect(() => {
+      const tabsContainer = tabsContainerRef.current
+      if (!tabsContainer) return
+
+      const updateScrollState = () => {
+        const {scrollLeft, scrollWidth, clientWidth} = tabsContainer
+        const hasOverflow = scrollWidth > clientWidth
+        const canScrollRight = hasOverflow && scrollLeft < scrollWidth - clientWidth - 1
+
+        tabsContainer.style.setProperty('--scroll-offset', `${scrollLeft}px`)
+        tabsContainer.style.setProperty('--has-overflow', hasOverflow ? '1' : '0')
+        tabsContainer.style.setProperty('--can-scroll-right', canScrollRight ? '1' : '0')
+      }
+
+      updateScrollState()
+      const resizeObserver = new ResizeObserver(updateScrollState)
+      resizeObserver.observe(tabsContainer)
+
+      // eslint-disable-next-line github/prefer-observers
+      tabsContainer.addEventListener('scroll', updateScrollState)
+
+      return () => {
+        tabsContainer.removeEventListener('scroll', updateScrollState)
+        resizeObserver.disconnect()
+      }
+    }, [tabs])
+
+    const tabListProps = getTabListProps({label: ariaLabel})
+
+    const handleNextTabControl = useCallback(() => {
+      if (activeTab && activeTab !== String(tabs.length - 1)) {
+        activateTab(String(Number(activeTab) + 1))
+      }
+    }, [activeTab, activateTab, tabs])
+
+    const handlePrevTabControl = useCallback(() => {
+      if (activeTab && activeTab !== '0') {
+        activateTab(String(Number(activeTab) - 1))
+      }
+    }, [activeTab, activateTab])
+
+    const hasMoreThanTwoTabs = tabs.length >= 3
+    const isLastTabActive = String(tabs.length - 1) === activeTab
+
+    return (
+      <div
+        className={clsx(styles['Tabs-container'], styles[`Tabs-container--align-${align}`], className)}
+        ref={ref}
+        data-testid={testIds.root}
+        {...props}
+      >
+        {/* Narrow only indicators and controls */}
+        {tabs.length >= 3 && (
+          <div className={styles.Tabs__controls} role="toolbar" aria-label={internalAccessibleLabels.controls}>
```

Comment:
Do you mind if I bring this to the a11y team to get their input on the role and any other suggested attributes/functionality? There are a few things I don't necessarily know the answer to and I'd like some expert input; e.g. the "toolbar" role, whether the next/prev buttons need `aria-disabled` when the first/last page is selected, etc.

I don't think it would need to block this PR

Reply:
I should note that, from looking at the [Toolbar Pattern APG](https://www.w3.org/WAI/ARIA/apg/patterns/toolbar/) it does feel to me that `"toolbar"` is an appropriate role.

To fully match the spec defined there we would also need to set an `aria-label` or `aria-labelledby` on the toolbar, and modify the focus behaviour to allow the prev/next controls to be navigated using the arrow keys.

Reply:
For posterity, the recommendation after speaking with the a11y team was to remove  `role="toolbar"` and `aria-label={internalAccessibleLabels.controls}`.